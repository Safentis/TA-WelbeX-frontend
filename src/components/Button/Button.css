.button {
    background-color: var(--main-dark-blue-color);
    box-shadow: .1rem .1rem .1rem var(--main-black-color);
    border: none;
    border-radius: .5rem;
    color: var(--main-white-color);
    cursor: pointer;
    padding: .5rem .75rem;
    outline: none;
    transition: .3s;
}

.button_active,
.button:hover {
    background-color: var(--main-dk-grey-color);
    box-shadow: .1rem .1rem .1rem var(--main-dk-grey-color);
}

button:hover {
    color: var(--main-dark-blue-color);
}

.button_active {
    color: var(--main-purple-color) !important;
}