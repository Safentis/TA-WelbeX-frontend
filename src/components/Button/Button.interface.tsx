export interface Props {
    style?: string;
    handleClick: () => void;
}