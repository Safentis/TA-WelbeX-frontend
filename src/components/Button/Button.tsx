import React, { FC } from 'react';

import './Button.css';
import { Props } from './Button.interface';

/**
 * Button component
 * @param {React.ReactElement | React.ReactNode | string} children - any children no required
 * @param {string} style - string style no required
 * @param {() => void} handleClick - function handleClick required
 * @returns {React.ReactElement} - returns react element
 */
const Button: FC<Props> = ({
  children,
  style = '',
  handleClick,
}): React.ReactElement => {
  return (
    <button className={`${style} button`} onClick={handleClick}>
      {children}
    </button>
  );
};

export default Button;
