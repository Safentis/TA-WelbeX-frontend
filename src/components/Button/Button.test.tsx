import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import Button from './Button';

import { Props } from './Button.interface';

const setUp = (props: Props) => shallow(<Button {...props} />);

describe('<Button />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;

  describe('Default props', () => {
    let props: Props = {
      handleClick: () => {},
      style: 'any-style',
    };

    beforeEach(() => {
      component = setUp(props);
    });

    it('component does match to the snapshot', () => {
      expect(component).toMatchSnapshot();
    });

    describe('Markup: ', () => {
      it('has 1 button element', () => {
        expect(component.find('button')).toBeTruthy();
      });
    });
  });

  describe('Custom props', () => {
    let props: Props = {
      handleClick: () => {},
    };

    beforeEach(() => {
      component = setUp(props);
    });

    it('component does match to the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
