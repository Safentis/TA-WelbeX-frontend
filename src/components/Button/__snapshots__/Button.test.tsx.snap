// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Custom props component does match to the snapshot 1`] = `
<button
  className=" button"
  onClick={[Function]}
/>
`;

exports[`<Button /> Default props component does match to the snapshot 1`] = `
<button
  className="any-style button"
  onClick={[Function]}
/>
`;
